django/contrib/gis/gdal/libgdal.py

```diff
@@ -83,23 +83,16 @@ def gdal_full_version():
     return _version_info(b'')
 
 
-version_regex = _lazy_re_compile(r'^(?P<major>\d+)\.(?P<minor>\d+)(\.(?P<subminor>\d+))?')
-
-
 def gdal_version_info():
-    ver = gdal_version().decode()
-    m = version_regex.match(ver)
+    ver = gdal_version()
+    m = re.match(br'^(?P<major>\d+)\.(?P<minor>\d+)(?:\.(?P<subminor>\d+))?', ver)
```

Comment:
I'd rename `subminor` to `patch`.

Reply:
We should also change `\d+` to `[0-9]+` in all cases.

Reply:
> We should also change `\d+` to `[0-9]+` in all cases.

Why?

Reply:
> Why?

`\d` will also [match any unicode decimal digit](https://docs.python.org/3/library/re.html#index-28), not only `0-9`. Granted, it is unlikely to cause a headache here.

Reply:
This is `bytes` pattern now.

Reply:
You're right. You know I both saw that and missed it too...

Reply:
@sir-sigurd Is there any reason to change a regex? :thinking: You added `:`.

Reply:
Yes. Adding `?:` makes it a non-capturing group. Which Allows for use of `m.groups()` below. Otherwise it'd need to be `... = m[1], m[2], m[4]`.

Reply:
I added [`?:`](https://docs.python.org/3/library/re.html#index-16), so `m.groups()` returns only the needed groups.

Reply:
Thanks :+1: